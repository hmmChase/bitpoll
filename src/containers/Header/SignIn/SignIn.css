.sign-in-btn:hover {
  background: #000;
}

.sign-in-btn {
  background: #3a3133;
  color: #fff;
  border: none;
  border-radius: 10px;
  height: 50px;
  display: flex;
  align-items: center;
  padding: 10px;
  margin: 20px;
}

.github-icon {
  margin-right: 5px;
  font-size: 1.5em;
}

.sign-in-btn- txt {
  margin-left: 5px;
}
