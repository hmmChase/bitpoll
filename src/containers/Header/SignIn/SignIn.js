import React from 'react';
import fontawesome from '@fortawesome/fontawesome';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import faGithub from '@fortawesome/fontawesome-free-brands';
import './SignIn.css';

const SignIn = props => {
  return (
    <div className="sign-in-btn" onClick={props.login}>
      <FontAwesomeIcon className="github-icon" icon={['fab', 'github']} />
      <span className="sign-in-btn-txt">Sign in with GitHub</span>
    </div>
  );
};

export default SignIn;
