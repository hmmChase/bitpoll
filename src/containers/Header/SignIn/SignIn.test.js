import React from 'react';
import { shallow } from 'enzyme';
import SignIn from './SignIn';

describe('SignIn', () => {
  let signIn;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      login: jest.fn()
    };
    signIn = shallow(<SignIn {...mockProps} />);
  });

  it('matches snapshot', () => {
    expect(signIn).toMatchSnapshot();
  });

  it('calls props.login on click', () => {
    signIn.find('.sign-in-btn').simulate('click');

    expect(mockProps.login).toHaveBeenCalledTimes(1);
  });
});
