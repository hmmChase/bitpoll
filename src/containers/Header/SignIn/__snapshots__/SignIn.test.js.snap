// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    login={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FontAwesomeIcon
          border={false}
          className="github-icon"
          fixedWidth={false}
          flip={null}
          icon={
                    Array [
                              "fab",
                              "github",
                            ]
          }
          listItem={false}
          mask={null}
          name=""
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          transform={null}
/>,
        <span
          className="sign-in-btn-txt"
>
          Sign in with GitHub
</span>,
      ],
      "className": "sign-in-btn",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "border": false,
          "className": "github-icon",
          "fixedWidth": false,
          "flip": null,
          "icon": Array [
            "fab",
            "github",
          ],
          "listItem": false,
          "mask": null,
          "name": "",
          "pull": null,
          "pulse": false,
          "rotation": null,
          "size": null,
          "spin": false,
          "symbol": false,
          "transform": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign in with GitHub",
          "className": "sign-in-btn-txt",
        },
        "ref": null,
        "rendered": "Sign in with GitHub",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FontAwesomeIcon
            border={false}
            className="github-icon"
            fixedWidth={false}
            flip={null}
            icon={
                        Array [
                                    "fab",
                                    "github",
                                  ]
            }
            listItem={false}
            mask={null}
            name=""
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            transform={null}
/>,
          <span
            className="sign-in-btn-txt"
>
            Sign in with GitHub
</span>,
        ],
        "className": "sign-in-btn",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "border": false,
            "className": "github-icon",
            "fixedWidth": false,
            "flip": null,
            "icon": Array [
              "fab",
              "github",
            ],
            "listItem": false,
            "mask": null,
            "name": "",
            "pull": null,
            "pulse": false,
            "rotation": null,
            "size": null,
            "spin": false,
            "symbol": false,
            "transform": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign in with GitHub",
            "className": "sign-in-btn-txt",
          },
          "ref": null,
          "rendered": "Sign in with GitHub",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
